packages/react-components/src/components/table-output-component.tsx

```diff
@@ -212,23 +219,59 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         this.setState({
             tableColumns: this.columnArray
         });
+        if (this.columnApi) {
+            const columns = this.columnApi.getAllColumns();
+            const timestampHeader = columns.find(column => column.getColDef().headerName === 'Timestamp ns');
+            if (timestampHeader) {
+                this.timestampCol = timestampHeader.getColDef().field;
+            }
+        }
+
     }
 
     private async handleTimeSelectionChange(range: TimelineChart.TimeGraphRange) {
         const start = Math.trunc(this.props.range.getstart() + range.start);
-        if (this.lastIndex.timestamp === start) {
+        const end = Math.trunc(this.props.range.getstart() + range.end);
+
+        this.startTimeStamp = Math.min(start, end);
+        this.endTimeStamp = Math.max(start, end);
+        const startIndex = await this.fetchTableIndex(start);
+
+        if (this.lastSelectedRange.startTimestamp === start &&
```

Comment:
We should do it a bit differently: If and only if the startTimestamp has changed, do the fetchTableIndex and ensure it's visible.

Then regardless of the above, always update the selected rows.

This will avoid many requests to the server while dragging a selection.

Reply:
Updated the code to only fetch the index from the server when the previous start time is different than the current start time.